x-pack/plugins/alerting/server/usage/lib/get_telemetry_from_kibana.ts

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  AggregationsSingleMetricAggregateBase,
+  AggregationsCardinalityAggregate,
+  AggregationsTermsAggregateBase,
+  AggregationsStringTermsBucketKeys,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import { AlertingUsage } from '../types';
+import { NUM_ALERTING_RULE_TYPES } from '../alerting_usage_collector';
+import { parseSimpleRuleTypeBucket } from './parse_simple_rule_type_bucket';
+
+interface Opts {
+  esClient: ElasticsearchClient;
+  kibanaIndex: string;
+  logger: Logger;
+}
+
+type GetTotalCountsResults = Pick<
+  AlertingUsage,
+  | 'count_total'
+  | 'count_by_type'
+  | 'throttle_time'
+  | 'schedule_time'
+  | 'throttle_time_number_s'
+  | 'schedule_time_number_s'
+  | 'connectors_per_alert'
+>;
+
+interface GetTotalCountInUseResults {
+  countTotal: number;
+  countByType: Record<string, number>;
+  countNamespaces: number;
+}
+
+export async function getTotalCountAggregations({
+  esClient,
+  kibanaIndex,
+  logger,
+}: Opts): Promise<GetTotalCountsResults> {
+  try {
+    const query = {
+      index: kibanaIndex,
+      size: 0,
+      body: {
+        query: {
+          bool: {
+            // Aggregate over all rule saved objects
+            filter: [{ term: { type: 'alert' } }],
+          },
+        },
+        runtime_mappings: {
+          rule_action_count: {
+            type: 'long',
+            script: {
```

Comment:
I wonder if we could replace these scripts with runtime fields, in a follow-on PR.  Guess it depends if there's a performance difference (maybe runtime fields are worse?) and whether it's worth it in general.  These scripts are easy enough to understand, so maybe not.